esp-hal/src/uart.rs

```diff
@@ -455,6 +494,10 @@ pub struct UartRx<'d, Dm> {
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 #[non_exhaustive]
 pub enum ConfigError {
+    /// The requested baud rate is not achievable.
+    UnachievableBaudrate,
```

Comment:
I forget if we can mark enum variants as unstable (cc: @bugadani) but as this is related to the unstable tolerance stuff, we should also mark this as unstable?

Reply:
It's impossible to mark enum variants as unstable. All we can do is cfg them away and return an Unknown/Other error instead.

Reply:
Gotcha so we can do what we do for spi dma error: https://github.com/MabezDev/esp-hal/blob/2ff28b14b58d3d726aafc0456143e5022b06f0e6/esp-hal/src/spi/mod.rs#L26-L28

and in `verify_baudrate` we can cfg away the `match config.baudrate_tolerance {` when unstable isn't enabled.